<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><title>R: Empirical analysis of digital gene expression data in R</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<link rel="stylesheet" type="text/css" href="R.css">
</head><body>
<h1> Empirical analysis of digital gene expression data in R
<img class="toplogo" src="../../../doc/html/logo.jpg" alt="[R logo]">
</h1>
<hr>
<div align="center">
<a href="../../../doc/html/packages.html"><img src="../../../doc/html/left.jpg" alt="[Up]" width="30" height="30" border="0"></a>
<a href="../../../doc/html/index.html"><img src="../../../doc/html/up.jpg" alt="[Top]" width="30" height="30" border="0"></a>
</div><h2>Documentation for package &lsquo;edgeR&rsquo; version 2.4.3</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
<li><a href="../doc/index.html">Overview of user guides and package vignettes</a>; browse <a href="../doc">directory</a>.</li>
<li><a href="../NEWS">Package NEWS</a>.</li>
</ul>

<h2>Help Pages</h2>


<p align="center">
<a href="# "> </a>
<a href="#A">A</a>
<a href="#B">B</a>
<a href="#C">C</a>
<a href="#D">D</a>
<a href="#E">E</a>
<a href="#G">G</a>
<a href="#L">L</a>
<a href="#M">M</a>
<a href="#N">N</a>
<a href="#P">P</a>
<a href="#Q">Q</a>
<a href="#R">R</a>
<a href="#S">S</a>
<a href="#T">T</a>
<a href="#W">W</a>
<a href="#misc">misc</a>
</p>

<table width="100%">
<tr><td width="25%"><a href="edgeR-package.html">edgeR-package</a></td>
<td>Empirical analysis of digital gene expression data in R</td></tr>
</table>

<h2><a name="A">-- A --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="adjustedProfileLik.html">adjustedProfileLik</a></td>
<td>Compute Cox-Reid Adjusted Profile Likelihood for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="approx.expected.info.html">approx.expected.info</a></td>
<td>Approximate Expected Information (Fisher Information)</td></tr>
<tr><td width="25%"><a href="asdataframe.html">as.data.frame.TopTags</a></td>
<td>Turn a TopTags Object into a Dataframe</td></tr>
<tr><td width="25%"><a href="asmatrix.html">as.matrix.DGEList</a></td>
<td>Turn a DGEList Object into a Matrix</td></tr>
</table>

<h2><a name="B">-- B --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="betaApproxNBTest.html">betaApproxNBTest</a></td>
<td>An Approximate Exact Test for Differences between Two Negative Binomial Groups</td></tr>
<tr><td width="25%"><a href="bin.dispersion.html">binCMLDispersion</a></td>
<td>Estimate Common Dispersion for Negative Binomial GLMs in Bins of Genes Sorted by Overall Abundance</td></tr>
<tr><td width="25%"><a href="bin.dispersion.html">binGLMDispersion</a></td>
<td>Estimate Common Dispersion for Negative Binomial GLMs in Bins of Genes Sorted by Overall Abundance</td></tr>
<tr><td width="25%"><a href="meanvar.html">binMeanVar</a></td>
<td>Explore the mean-variance relationship for DGE data</td></tr>
<tr><td width="25%"><a href="binomTest.html">binomTest</a></td>
<td>Exact Binomial Tests for Comparing Two Digital Libraries</td></tr>
</table>

<h2><a name="C">-- C --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="calcNormFactors.html">calcNormFactors</a></td>
<td>Calculates Normalization Factors for a Matrix of Count Data</td></tr>
<tr><td width="25%"><a href="commonCondLogLikDerDelta.html">commonCondLogLikDerDelta</a></td>
<td>Conditional Log-Likelihoods in Terms of Delta</td></tr>
<tr><td width="25%"><a href="condLogLikDerDelta.html">condLogLikDerDelta</a></td>
<td>Conditional Log-Likelihood in Terms of Delta</td></tr>
<tr><td width="25%"><a href="condLogLikDerSize.html">condLogLikDerSize</a></td>
<td>Log-Likelihood of the Common Dispersion for a Single Equalized Group</td></tr>
<tr><td width="25%"><a href="cpm.html">cpm</a></td>
<td>Calculate Counts per Million from DGEList or Matrix Object</td></tr>
<tr><td width="25%"><a href="cutWithMinN.html">cutWithMinN</a></td>
<td>Cut numeric vector into non-empty intervals</td></tr>
</table>

<h2><a name="D">-- D --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="decidetestsDGE.html">decideTestsDGE</a></td>
<td>Multiple Testing Across Genes and Contrasts</td></tr>
<tr><td width="25%"><a href="mglm.html">designAsFactor</a></td>
<td>Fit Negative Binomial Generalized Linear Model to Multiple Response Vectors</td></tr>
<tr><td width="25%"><a href="mglm.html">deviances.function</a></td>
<td>Fit Negative Binomial Generalized Linear Model to Multiple Response Vectors</td></tr>
<tr><td width="25%"><a href="DGEExact-class.html">DGEExact-class</a></td>
<td>differential expression of Digital Gene Expression data - class</td></tr>
<tr><td width="25%"><a href="DGEGLM.html">DGEGLM-class</a></td>
<td>Digital Gene Expression Generalized Linear Model results - class</td></tr>
<tr><td width="25%"><a href="DGEList.html">DGEList</a></td>
<td>DGEList Constructor</td></tr>
<tr><td width="25%"><a href="DGEList-class.html">DGEList-class</a></td>
<td>Digital Gene Expression data - class</td></tr>
<tr><td width="25%"><a href="DGELRT.html">DGELRT-class</a></td>
<td>Digital Gene Expression Likelihood Ratio Test data and results - class</td></tr>
<tr><td width="25%"><a href="dglmStdResid.html">dglmStdResid</a></td>
<td>Visualize the mean-variance relationship in DGE data using standardized residuals</td></tr>
<tr><td width="25%"><a href="dim.html">dim.DGEExact</a></td>
<td>Retrieve the Dimensions of a DGEList, DGEExact, DGEGLM, DGELRT or TopTags Object</td></tr>
<tr><td width="25%"><a href="dim.html">dim.DGEGLM</a></td>
<td>Retrieve the Dimensions of a DGEList, DGEExact, DGEGLM, DGELRT or TopTags Object</td></tr>
<tr><td width="25%"><a href="dim.html">dim.DGEList</a></td>
<td>Retrieve the Dimensions of a DGEList, DGEExact, DGEGLM, DGELRT or TopTags Object</td></tr>
<tr><td width="25%"><a href="dim.html">dim.DGELRT</a></td>
<td>Retrieve the Dimensions of a DGEList, DGEExact, DGEGLM, DGELRT or TopTags Object</td></tr>
<tr><td width="25%"><a href="dim.html">dim.TopTags</a></td>
<td>Retrieve the Dimensions of a DGEList, DGEExact, DGEGLM, DGELRT or TopTags Object</td></tr>
<tr><td width="25%"><a href="dimnames.html">dimnames.DGEList</a></td>
<td>Retrieve the Dimension Names of a DGEList Object</td></tr>
<tr><td width="25%"><a href="dimnames.html">dimnames&lt;-.DGEList</a></td>
<td>Retrieve the Dimension Names of a DGEList Object</td></tr>
<tr><td width="25%"><a href="dispBinTrend.html">dispBinTrend</a></td>
<td>Estimate Dispersions with an Abundance-Dependent Trend for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="dispCoxReid.html">dispCoxReid</a></td>
<td>Estimate Common Dispersion for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="dispCoxReidInterpolateTagwise.html">dispCoxReidInterpolateTagwise</a></td>
<td>Estimate Tagwise Dispersion for Negative Binomial GLMs by Cox-Reid Adjusted Profile Likelihood</td></tr>
<tr><td width="25%"><a href="dispCoxReidSplineTrend.html">dispCoxReidPowerTrend</a></td>
<td>Estimate Dispersion Trend for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="dispCoxReidSplineTrend.html">dispCoxReidSplineTrend</a></td>
<td>Estimate Dispersion Trend for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="dispCoxReid.html">dispDeviance</a></td>
<td>Estimate Common Dispersion for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="dispCoxReid.html">dispPearson</a></td>
<td>Estimate Common Dispersion for Negative Binomial GLMs</td></tr>
</table>

<h2><a name="E">-- E --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="edgeR-package.html">edgeR</a></td>
<td>Empirical analysis of digital gene expression data in R</td></tr>
<tr><td width="25%"><a href="equalizeLibSizes.html">equalizeLibSizes</a></td>
<td>Quantile Adjustment to Equalize Library Sizes for a Fixed Value of the Dispersion Parameter</td></tr>
<tr><td width="25%"><a href="estimateCommonDisp.html">estimateCommonDisp</a></td>
<td>Estimates the Negative Binomial Common Dispersion by Maximizing the Negative Binomial Conditional Common Likelihood</td></tr>
<tr><td width="25%"><a href="estimateExonGenewisedisp.html">estimateExonGenewiseDisp</a></td>
<td>Estimate Genewise Dispersions from Exon-Level Count Data</td></tr>
<tr><td width="25%"><a href="estimateGLMCommonDisp.html">estimateGLMCommonDisp</a></td>
<td>Estimate Common Dispersion for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="estimateGLMCommonDisp.html">estimateGLMCommonDisp.default</a></td>
<td>Estimate Common Dispersion for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="estimateGLMCommonDisp.html">estimateGLMCommonDisp.DGEList</a></td>
<td>Estimate Common Dispersion for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="estimateGLMTagwiseDisp.html">estimateGLMTagwiseDisp</a></td>
<td>Estimate Empirical Bayes Tagwise Dispersions for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="estimateGLMTagwiseDisp.html">estimateGLMTagwiseDisp.default</a></td>
<td>Estimate Empirical Bayes Tagwise Dispersions for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="estimateGLMTagwiseDisp.html">estimateGLMTagwiseDisp.DGEList</a></td>
<td>Estimate Empirical Bayes Tagwise Dispersions for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="estimateGLMTrendedDisp.html">estimateGLMTrendedDisp</a></td>
<td>Estimate Trended Dispersion for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="estimateGLMTrendedDisp.html">estimateGLMTrendedDisp.default</a></td>
<td>Estimate Trended Dispersion for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="estimateGLMTrendedDisp.html">estimateGLMTrendedDisp.DGEList</a></td>
<td>Estimate Trended Dispersion for Negative Binomial GLMs</td></tr>
<tr><td width="25%"><a href="estimatePs.html">estimatePs</a></td>
<td>Estimate Expression Levels</td></tr>
<tr><td width="25%"><a href="estimateSmoothing.html">estimateSmoothing</a></td>
<td>Estimate the Prior Weight for Tagwise Dispersions</td></tr>
<tr><td width="25%"><a href="estimateTagwiseDisp.html">estimateTagwiseDisp</a></td>
<td>Estimate Empirical Bayes Tagwise Dispersion Values</td></tr>
<tr><td width="25%"><a href="exactTest.html">exactTest</a></td>
<td>Exact Tests for Differences between Two Groups of Negative-Binomial Counts</td></tr>
<tr><td width="25%"><a href="exactTest.html">exactTestBetaApprox</a></td>
<td>Exact Tests for Differences between Two Groups of Negative-Binomial Counts</td></tr>
<tr><td width="25%"><a href="exactTest.html">exactTestByDeviance</a></td>
<td>Exact Tests for Differences between Two Groups of Negative-Binomial Counts</td></tr>
<tr><td width="25%"><a href="exactTest.html">exactTestBySmallP</a></td>
<td>Exact Tests for Differences between Two Groups of Negative-Binomial Counts</td></tr>
<tr><td width="25%"><a href="exactTest.html">exactTestDoubleTail</a></td>
<td>Exact Tests for Differences between Two Groups of Negative-Binomial Counts</td></tr>
<tr><td width="25%"><a href="expandAsMatrix.html">expandAsMatrix</a></td>
<td>expandAsMatrix</td></tr>
</table>

<h2><a name="G">-- G --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="getCounts.html">getCounts</a></td>
<td>Extract Table of Counts from DGEList Object</td></tr>
<tr><td width="25%"><a href="dglmStdResid.html">getDispersions</a></td>
<td>Visualize the mean-variance relationship in DGE data using standardized residuals</td></tr>
<tr><td width="25%"><a href="getOffset.html">getOffset</a></td>
<td>Extract Vector of Offsets from DGEList Object</td></tr>
<tr><td width="25%"><a href="getPriorN.html">getPriorN</a></td>
<td>Get a Recommended Value for Prior N from DGEList Object</td></tr>
<tr><td width="25%"><a href="glmfit.html">glmFit</a></td>
<td>Fit negative binomial generalized linear model for each transcript</td></tr>
<tr><td width="25%"><a href="glmfit.html">glmFit.default</a></td>
<td>Fit negative binomial generalized linear model for each transcript</td></tr>
<tr><td width="25%"><a href="glmfit.html">glmFit.DGEList</a></td>
<td>Fit negative binomial generalized linear model for each transcript</td></tr>
<tr><td width="25%"><a href="glmfit.html">glmLRT</a></td>
<td>Fit negative binomial generalized linear model for each transcript</td></tr>
<tr><td width="25%"><a href="gof.html">gof</a></td>
<td>Goodness of Fit Tests for Multiple GLM Fits</td></tr>
<tr><td width="25%"><a href="goodTuring.html">goodTuring</a></td>
<td>Good-Turing Frequency Estimation</td></tr>
<tr><td width="25%"><a href="goodTuring.html">goodTuringProportions</a></td>
<td>Good-Turing Frequency Estimation</td></tr>
</table>

<h2><a name="L">-- L --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="dim.html">length.DGEExact</a></td>
<td>Retrieve the Dimensions of a DGEList, DGEExact, DGEGLM, DGELRT or TopTags Object</td></tr>
<tr><td width="25%"><a href="dim.html">length.DGEGLM</a></td>
<td>Retrieve the Dimensions of a DGEList, DGEExact, DGEGLM, DGELRT or TopTags Object</td></tr>
<tr><td width="25%"><a href="dim.html">length.DGEList</a></td>
<td>Retrieve the Dimensions of a DGEList, DGEExact, DGEGLM, DGELRT or TopTags Object</td></tr>
<tr><td width="25%"><a href="dim.html">length.DGELRT</a></td>
<td>Retrieve the Dimensions of a DGEList, DGEExact, DGEGLM, DGELRT or TopTags Object</td></tr>
<tr><td width="25%"><a href="dim.html">length.TopTags</a></td>
<td>Retrieve the Dimensions of a DGEList, DGEExact, DGEGLM, DGELRT or TopTags Object</td></tr>
<tr><td width="25%"><a href="logLikDerP.html">logLikDerP</a></td>
<td>Log-Likelihood for Proportion</td></tr>
</table>

<h2><a name="M">-- M --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="maPlot.html">maPlot</a></td>
<td>Plots Log-Fold Change versus Log-Concentration (or, M versus A) for Count Data</td></tr>
<tr><td width="25%"><a href="maximizeInterpolant.html">maximizeInterpolant</a></td>
<td>Maximize a function given a table of values by spline interpolation.</td></tr>
<tr><td width="25%"><a href="mglm.html">mglm</a></td>
<td>Fit Negative Binomial Generalized Linear Model to Multiple Response Vectors</td></tr>
<tr><td width="25%"><a href="mglm.html">mglmLevenberg</a></td>
<td>Fit Negative Binomial Generalized Linear Model to Multiple Response Vectors</td></tr>
<tr><td width="25%"><a href="mglm.html">mglmLS</a></td>
<td>Fit Negative Binomial Generalized Linear Model to Multiple Response Vectors</td></tr>
<tr><td width="25%"><a href="mglm.html">mglmOneGroup</a></td>
<td>Fit Negative Binomial Generalized Linear Model to Multiple Response Vectors</td></tr>
<tr><td width="25%"><a href="mglm.html">mglmOneWay</a></td>
<td>Fit Negative Binomial Generalized Linear Model to Multiple Response Vectors</td></tr>
<tr><td width="25%"><a href="mglm.html">mglmSimple</a></td>
<td>Fit Negative Binomial Generalized Linear Model to Multiple Response Vectors</td></tr>
<tr><td width="25%"><a href="movingAverageByCol.html">movingAverageByCol</a></td>
<td>Moving Average Smoother of Matrix Columns</td></tr>
</table>

<h2><a name="N">-- N --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="Tu102.html">NC1</a></td>
<td>Raw Data for Several SAGE Libraries from the Zhang 1997 Science Paper.</td></tr>
<tr><td width="25%"><a href="Tu102.html">NC2</a></td>
<td>Raw Data for Several SAGE Libraries from the Zhang 1997 Science Paper.</td></tr>
</table>

<h2><a name="P">-- P --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="plotExonUsage.html">plotExonUsage</a></td>
<td>Create a Plot of Exon Usage from Exon-Level Count Data</td></tr>
<tr><td width="25%"><a href="plotMDS.dge.html">plotMDS.DGEList</a></td>
<td>Multidimensional scaling plot of digital gene expression profiles</td></tr>
<tr><td width="25%"><a href="meanvar.html">plotMeanVar</a></td>
<td>Explore the mean-variance relationship for DGE data</td></tr>
<tr><td width="25%"><a href="plotSmear.html">plotSmear</a></td>
<td>Plots log-Fold Change versus log-Concentration (or, M versus A) for Count Data</td></tr>
</table>

<h2><a name="Q">-- Q --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="q2qnbinom.html">q2qnbinom</a></td>
<td>Quantile to Quantile Mapping between Negative-Binomial Distributions</td></tr>
<tr><td width="25%"><a href="q2qnbinom.html">q2qpois</a></td>
<td>Quantile to Quantile Mapping between Negative-Binomial Distributions</td></tr>
</table>

<h2><a name="R">-- R --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="readDGE.html">readDGE</a></td>
<td>Read and Merge a Set of Files Containing DGE Data</td></tr>
</table>

<h2><a name="S">-- S --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="DGEExact-class.html">show-method</a></td>
<td>differential expression of Digital Gene Expression data - class</td></tr>
<tr><td width="25%"><a href="DGEGLM.html">show-method</a></td>
<td>Digital Gene Expression Generalized Linear Model results - class</td></tr>
<tr><td width="25%"><a href="DGELRT.html">show-method</a></td>
<td>Digital Gene Expression Likelihood Ratio Test data and results - class</td></tr>
<tr><td width="25%"><a href="topTags.html">show-method</a></td>
<td>Table of the Top Differentially Expressed Tags</td></tr>
<tr><td width="25%"><a href="spliceVariants.html">spliceVariants</a></td>
<td>Identify Genes with Splice Variants</td></tr>
<tr><td width="25%"><a href="splitIntoGroups.html">splitIntoGroups</a></td>
<td>Split the Counts or Pseudocounts from a DGEList Object According To Group</td></tr>
<tr><td width="25%"><a href="splitIntoGroups.html">splitIntoGroupsPseudo</a></td>
<td>Split the Counts or Pseudocounts from a DGEList Object According To Group</td></tr>
<tr><td width="25%"><a href="subsetting.html">subsetting</a></td>
<td>Subset DGEList, DGEGLM, DGEExact and DGELRT Objects</td></tr>
<tr><td width="25%"><a href="systematicSubset.html">systematicSubset</a></td>
<td>Take a systematic subset of indices.</td></tr>
</table>

<h2><a name="T">-- T --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="thinCounts.html">thinCounts</a></td>
<td>Binomial Thinning of Counts</td></tr>
<tr><td width="25%"><a href="topTags.html">topTags</a></td>
<td>Table of the Top Differentially Expressed Tags</td></tr>
<tr><td width="25%"><a href="topTags.html">TopTags-class</a></td>
<td>Table of the Top Differentially Expressed Tags</td></tr>
<tr><td width="25%"><a href="Tu102.html">Tu102</a></td>
<td>Raw Data for Several SAGE Libraries from the Zhang 1997 Science Paper.</td></tr>
<tr><td width="25%"><a href="Tu102.html">Tu98</a></td>
<td>Raw Data for Several SAGE Libraries from the Zhang 1997 Science Paper.</td></tr>
</table>

<h2><a name="W">-- W --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="weightedComLik.html">weightedComLik</a></td>
<td>Weighted Common Log-Likelihood</td></tr>
<tr><td width="25%"><a href="weightedComLik.html">weightedComLikMA</a></td>
<td>Weighted Common Log-Likelihood</td></tr>
<tr><td width="25%"><a href="weightedCondLogLikDerDelta.html">weightedCondLogLikDerDelta</a></td>
<td>Weighted Conditional Log-Likelihood in Terms of Delta</td></tr>
</table>

<h2><a name="misc">-- misc --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="subsetting.html">[.DGEExact</a></td>
<td>Subset DGEList, DGEGLM, DGEExact and DGELRT Objects</td></tr>
<tr><td width="25%"><a href="subsetting.html">[.DGEGLM</a></td>
<td>Subset DGEList, DGEGLM, DGEExact and DGELRT Objects</td></tr>
<tr><td width="25%"><a href="subsetting.html">[.DGEList</a></td>
<td>Subset DGEList, DGEGLM, DGEExact and DGELRT Objects</td></tr>
<tr><td width="25%"><a href="subsetting.html">[.DGELRT</a></td>
<td>Subset DGEList, DGEGLM, DGEExact and DGELRT Objects</td></tr>
<tr><td width="25%"><a href="topTags.html">[.TopTags</a></td>
<td>Table of the Top Differentially Expressed Tags</td></tr>
</table>
</body></html>
